src/Auth.js

```diff
@@ -27,6 +30,65 @@ function Auth({
   this.rolePromise = null;
 }
 
+function base64url(source) {
+  var encodedSource = CryptoJS.enc.Base64.stringify(source);
+
+  encodedSource = encodedSource.replace(/=+$/, '');
+
+  encodedSource = encodedSource.replace(/\+/g, '-');
+  encodedSource = encodedSource.replace(/\//g, '_');
+
+  return encodedSource;
+}
+
+const generateRefreshToken = function () {
+  return SHA256(CryptoJS.lib.WordArray.random(256)).toString();
+};
+
+const createJWT = function (sessionToken, oauthKey, oauthTTL) {
+  const header = {
+    alg: 'HS256',
+    typ: 'JWT',
+  };
+
+  const stringifiedHeader = CryptoJS.enc.Utf8.parse(JSON.stringify(header));
+  const encodedHeader = base64url(stringifiedHeader);
+
+  const timestamp = Math.floor(new Date().getTime() / 1000);
+  const expiration = timestamp + oauthTTL;
```

Comment:
I see, you are doing `timestamp + ttl`. `expiresAt` seems appropriate IMO

Reply:
Done :)